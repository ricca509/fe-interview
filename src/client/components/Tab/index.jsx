import React from 'react';

const Tab = ({ title, tabIndex, onClick, isSelected = false }) => {
  const onTabClick = event => {
    event.preventDefault();
    onClick(tabIndex);
  };

  return (
    <li className="tab">
      <a href="#" onClick={onTabClick}>
        {title}
      </a>
    </li>
  );
};

export default Tab;