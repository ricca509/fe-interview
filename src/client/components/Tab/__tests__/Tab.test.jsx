import React from 'react';
import Tab from '../';
import { shallow } from 'enzyme';

describe('Tab component', ()=> {
  const props = {
    title: 'Tab1',
    tabIndex: 0,
    onClick: ()=>({}),
    isSelected: true
  };

  it('should render a list element with the right title', () => {
    const wrapper = shallow(<Tab {...props}>content</Tab>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render with right props when `isSelected`', () => {
    const wrapper = shallow(<Tab {...props}>content</Tab>);

    expect(wrapper.find('li').prop('className')).toEqual('tab tab-selected');
  });

  it('should call the `onClick` handler correctly when clicked', () => {
    const ownProps = {
      ...props,
      onClick: jest.fn(),
    };
    const preventDefaultMock = jest.fn();

    const wrapper = shallow(<Tab {...ownProps}>content</Tab>);
    wrapper.find('a').simulate('click', {
      preventDefault: preventDefaultMock
    });

    expect(ownProps.onClick).toHaveBeenCalledWith(0);
    expect(preventDefaultMock).toHaveBeenCalled();
  });
});