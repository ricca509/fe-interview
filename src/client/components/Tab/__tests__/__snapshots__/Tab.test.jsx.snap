// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab component should render a list element with the right title 1`] = `
<li
  className="tab"
>
  <a
    href="#"
    onClick={[Function]}
  >
    Tab1
  </a>
</li>
`;
